tests/tasks/test_employee_appointments.py

```diff
@@ -0,0 +1,91 @@
+# ruff: noqa: PLR2004, PD901
+
+
+def test_extract_dw_employees_load_sql_query(
+    task_extract_dw_employee_appointment_complete,
+):
+    assert (
+        task_extract_dw_employee_appointment_complete.sql_file
+        == "hrqb/tasks/sql/employee_appointments.sql"
+    )
+    assert task_extract_dw_employee_appointment_complete.sql_query is not None
+
+
+def test_task_extract_qb_libhr_complete_has_required_fields(
+    task_extract_qb_libhr_complete,
+):
+    required_columns = [
+        "Related Employee MIT ID",
+        "Related Supervisor MIT ID",
+        "HC ID",
+        "Position ID",
+        "Cost Object",
+        "Related Department ID",
+    ]
+    assert set(required_columns).issubset(
+        set(task_extract_qb_libhr_complete.target.read().columns)
+    )
+
+
+def test_task_extract_qb_departments_complete_has_required_fields(
+    task_extract_qb_departments_complete,
+):
+    required_columns = ["Directorate", "Record ID#"]
+    assert set(required_columns).issubset(
+        set(task_extract_qb_departments_complete.target.read().columns)
+    )
+
+
+def test_task_transform_employee_appointments_libhr_data_merge_success(
+    task_transform_employee_appointments_complete,
+):
+    row = task_transform_employee_appointments_complete.get_dataframe().iloc[0]
+    assert row["HC ID"] == "L-001"
+    assert row["Cost Object"] == 7777777
+    assert row["Supervisor"] == "111111111"
+    assert row["Related Department ID"] == 42.0
```

Comment:
[Non-blocking] As I was reviewing these `_merge` tests, I initially thought it'd be helpful to include a check that were only merging on values that appear on the **left** dataframe (since `how="left"`) (i.e., by also asserting row counts). However, I started to wonder whether that would be attempting to test `pd.merge`, which doesn't seem like the thing to do. Just wanted to share my thoughts!

Reply:
Appreciate the pondering!  I'm finding testing in this project difficult, particularly at this grain of actual ETL tests that are dealing with data warehouse, quickbase, and sensitive data.

I think you're on to something about row counts, because it _is_ in `pd.merge()` where we'd likely introduce duplicate rows.

I should have commented on this somewhere, but I'm envisioning some "integrity" checks once things have settled (and getting close now, with only a couple more tables to add ETL tasks for).  I think combing through some of these and adding checks on row counts, duplicates, etc., would be a nice integrity test when tasks are actually running.  And, they could be confirmed with mocked unit test that would trigger duplicate rows in the merges as-is.  

All said, it's on the brain!  And I think it is somewhere in-between, like we need to think:
  * I don't want duplicate rows in task XYZ, because ABC
  * I have a runtime assertion in task XYZ that raises errors if that happens
  * I can write a test that simulates that and confirm runtime assertion works

Reply:
Sounds like a plan! Thank you for sharing. :)